src/Test/AbstractConfigTestCase.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Test;
+
+use PhpCsFixer\ConfigInterface;
+use PhpCsFixer\Fixer\DeprecatedFixerInterface;
+use PhpCsFixer\Fixer\FixerInterface;
+use PhpCsFixer\FixerFactory;
+use PhpCsFixer\RuleSet;
+use PHPUnit\Framework\TestCase;
+
+abstract class AbstractConfigTestCase extends TestCase
+{
+    final protected function doTestAllDefaultRulesAreSpecified(ConfigInterface $config)
```

Comment:
to be honest i got lost what this method does

Reply:
I'm sorry: I've entirely refactored the class extracting all the methods I could, now it should be straightforward to follow its logic.